R/otp-setup.R

```diff
@@ -0,0 +1,290 @@
+    }
+    set_up <- try(system(text, intern = FALSE, wait = FALSE))
+
+
+  }else if(Sys.info()[['sysname']] == "Darwin"){
+    message("You're on Mac, this function is not yet supported")
+    stop()
+
+  }else{
+    message("You're on and unknow OS, this function is not yet supported")
+    stop()
+
+  }
+
+
+  # Check for errors
+  if(grepl("ERROR",set_up[2])){
+    message("Failed to build graph with message:")
+    message(set_up[2])
+  }
+
+  message(paste0(Sys.time()," OTP is loading and may take a while to be useable"))
+
+  if(wait){
+    Sys.sleep(30)
+
+    # Check if connected
+    for(i in 1:10){
+      #message(paste0("Attempt ",i))
+      otpcon <- try(otp_connect(hostname = "localhost",
+                                router = router,
+                                port = port,
+                                ssl = FALSE,
+                                check = TRUE), silent = T)
+      if("otpconnect" %in% class(otpcon)){
+        message(paste0(Sys.time()," OTP is ready to use Go to localhost:",port," in your browser to view the OTP"))
+        browseURL(paste0(ifelse(otpcon$ssl,"https://","http://"),"localhost:",port))
+        break
+      }else{
+        if(i < 10){
+          Sys.sleep(30)
+        }else{
+          message(paste0(Sys.time()," OTP is taking an unusually long time to load, releasing R to your control"))
+        }
+
+      }
+    }
+  }
+
+}
+
+#' Stop and OTP Instance
+#'
+#' OTP is run in Java and requires Java commands to be typed into the command line.
+#' The function allows the parameters to be defined in R and automatically passed to Java.
+#' This function stops an already running OTP instance
+#'
+#' The function assumes you have run otp_setup()
+#'
+#' @export
+otp_stop <- function()
+{
+  if(Sys.info()[['sysname']] == "Linux") {
+    message("You're on linux, this function is not yet supported")
+  }else if(Sys.info()[['sysname']] == "Windows"){
+    readline(prompt="This will force Java to close, Press [enter] to continue, [escape] to abort")
+    system("Taskkill /IM java.exe /F", intern = TRUE)
+  }else if(Sys.info()[['sysname']] == "Darwin"){
+    message("You're on Mac, this function is not yet supported")
+  }else{
+    message("You're on and unknow OS, this function is not yet supported")
+  }
+
+}
+
+
+#' Basic OTP Setup Checks
+#'
+#' Checks to run before setting up the OTP
+#'
+#' @param dir A character string path to a folder containing the necessary files, see details
+#' @param router A character string for the name of the router, must match with contents of dir, default "current"
+#' @param graph Logical, check for graph, default = FALSE
+#'
+otp_checks <- function(otp = NULL, dir = NULL, router = NULL, graph = FALSE)
+{
+  ### Checks
+  # Check we have the directory defined
+  if(!exists("otp")){
+    warning("Path to the Open Trip Planner is not defined")
+    stop()
+  }
+  if(!exists("dir")){
+    warning("Path to the files to build graph are not defined")
+    stop()
+  }
```

Comment:
@mem48

These checks aren't doing as expected. otp and dir will always exist as `otp = null` etc creates an empty object. Want to be checking !is.null() ?